themes/default/content/what-is/run-aws-cloudwatch-get-metric-data-with-dynamic-credentials.md

```diff
@@ -0,0 +1,212 @@
+
+- **Seamless Command Execution** - The `esc run` command lets you execute AWS commands effortlessly, freeing you from the intricacies of managing AWS credentials on your local machine. Simply put, it significantly reduces the overhead of credential setup and maintenance.
+
+- **Enhanced Security** - One of the standout features of `esc run` is its commitment to security. Removing the local storage of credentials reduces the risk of accidental exposure. Your credentials and secrets are securely managed within the Pulumi environment.
+
+- **Streamlined Collaboration** - Because credentials are centralized, `esc run` facilitates smoother team collaboration by providing a consistent environment for all team members to leverage. Everyone can access the same secure environment, reducing the complexities of coordinating credentials and configurations across teams.
+
+## Getting started with esc run
+
+### Step 1: Install and login to Pulumi ESC
+
+To begin, you’ll need to [install Pulumi ESC](/docs/install/esc/). Once the installation is complete, run the `esc login` command and follow the steps to log in to the CLI.
+
+```bash
+$ esc login
+Manage your Pulumi ESC environments by logging in.
+Run `esc --help` for alternative login options.
+Enter your access token from https://app.pulumi.com/account/tokens
+    or hit <ENTER> to log in using your browser                   :  
+Logged in to pulumi.com as …
+```
+
+### Step 2: Create the OIDC configuration
+
+Pulumi ESC allows you to manually set your credentials as secrets in your Pulumi ESC environment files. When it comes to something like OIDC configuration, a more secure and efficient alternative is to leverage yet another great feature of Pulumi ESC: dynamic credentials.
+
+This service can dynamically generate credentials on your behalf whenever you interact with your AWS environments. To do so, follow the [guide for configuring OIDC between Pulumi and AWS](/docs/pulumi-cloud/oidc/aws/). Ensure the IAM role you create has sufficient permissions to perform the AWS CloudWatch actions.
+
+### Step 3: Create a new Pulumi ESC environment
+
+Once you have OIDC configured between Pulumi and AWS, the next step is to create a new environment in [Pulumi Cloud](https://app.pulumi.com/).
+
+- Navigate to your [Pulumi Cloud](https://app.pulumi.com/) home page.
+- Select the correct organization in the left-hand navigation menu.
+- Click the **Environments** link.
+- Click the **Create environment** button.
+- In the pop-up window, provide a name for your environment. e.g., `aws-prod-env`
+- Click the **Create environment** button.
+
+{{< video title="Open environment in Pulumi ESC console" src="/what-is/esc-create-new-env.mp4" autoplay="true" loop="true" >}}
```

Comment:
That's correct, it's been added under another PR before this one gets merged.

Reply:
Which PR, if you don't mind me asking? (I'll go review that PR as well.)

Reply:
hey @desteves can you add the video to this PR; yours will be the first to go out so the rest that follow can use it.

Reply:
I thought it was [this one](https://github.com/pulumi/pulumi-hugo-private/pull/189/files). I have been making updates there because James said he'd incorporated Aaron's feedback on it.